lib/resourcebuilder/apps.go

```diff
@@ -54,9 +56,11 @@ func (b *deploymentBuilder) Do(ctx context.Context) error {
 	return nil
 }
 func waitForDeploymentCompletion(ctx context.Context, client appsclientv1.DeploymentsGetter, deployment *appsv1.Deployment) error {
-	return wait.PollImmediateUntil(defaultObjectPollInterval, func() (bool, error) {
-		d, err := client.Deployments(deployment.Namespace).Get(deployment.Name, metav1.GetOptions{})
-		if errors.IsNotFound(err) {
+	var d *appsv1.Deployment
```

Comment:
instead of keeping global d. Please use lastErr that stores the last known error in the Poll loop. keeping the error handling confined... example operatorstatus.go